src/libmongoc/CMakeLists.txt

```diff
@@ -78,7 +75,7 @@ endif ()
 if ( (ENABLE_ZLIB STREQUAL "BUNDLED")
    OR (ENABLE_ZLIB STREQUAL "AUTO" AND NOT ZLIB_FOUND) )
    message (STATUS "Enabling zlib compression (bundled)")
-   set (SOURCES ${SOURCES} ${ZLIB_SOURCES})
+   set (SOURCES ${SOURCES} $<TARGET_OBJECTS:zlib_obj>)
```

Comment:
It's not clear to me that this change is necessary or correct.  From the `cmake-generator-expressions` man page:

```
$<TARGET_OBJECTS:objLib>
              List of objects resulting from build of objLib. objLib must be an object of type OBJECT_LIBRARY.
```

That seems to indicate that the generator expression will actually provide a list of `.o` files.  Am I mistaken in thinking this?

Reply:
This is correct. `TARGET_OBJECTS` is designed to be used in the source lists of other targets. Using `TARGET_OBJECTS` will have the object files from zlib directly included into the library, but they will compiled under the rules of the `zlib_obj` library. This prevents the zlib sources from being compiled with the flags that we use on the mongoc targets (e.g. the warnings flags), but they'll still be included in the mongoc libraries directly.

Reply:
Cool!  TIL